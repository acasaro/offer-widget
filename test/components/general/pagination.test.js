import React from 'react';
import { mount, shallow } from 'enzyme';
import Pagination from '../../../src/components/general/pagination/pagination';
import Page from '../../../src/components/general/pagination/page';

describe('Pagination Component', () => {
	it('should render a pagination component', () => {
		const mockPages = [{ key: 'test1' }, { key: 'test2' }];
		const pagination = mount(<Pagination pages={mockPages} activePage="test1" handlePageChange={() => {}} />);
		expect(pagination).toMatchSnapshot();
		expect(pagination.find(Page).find('.is-active')).toHaveLength(1);
	});

	it('should still render a pagination component with missing props', () => {
		const pagination = shallow(<Pagination handlePageChange={() => {}} />);
		expect(pagination).toMatchSnapshot();
	});
});
