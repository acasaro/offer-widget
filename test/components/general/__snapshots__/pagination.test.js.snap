// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component should render a pagination component 1`] = `
<Pagination
  activePage="test1"
  handlePageChange={[Function]}
  pages={
    Array [
      Object {
        "key": "test1",
      },
      Object {
        "key": "test2",
      },
    ]
  }
>
  <div
    className="pagination"
  >
    <Page
      handlePageChange={[Function]}
      isActive={true}
      key="test1"
      page={
        Object {
          "key": "test1",
        }
      }
    >
      <button
        className="button paging-circle is-active"
        onClick={[Function]}
      />
    </Page>
    <Page
      handlePageChange={[Function]}
      isActive={false}
      key="test2"
      page={
        Object {
          "key": "test2",
        }
      }
    >
      <button
        className="button paging-circle"
        onClick={[Function]}
      />
    </Page>
  </div>
</Pagination>
`;

exports[`Pagination Component should still render a pagination component with missing props 1`] = `
<div
  className="pagination"
/>
`;
